src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,17 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (!is_scalar($tagValue)
+                            && !(is_object($tagValue) && method_exists($tagValue, '__toString'))
```

Comment:
Me neither but we are not using classes for tags nor logs so this is the best we can do. What do you think @beberlei ?